client/shared/src/search/parser/parser.ts

```diff
@@ -272,6 +272,122 @@ const scanToken = <T extends Term = Literal>(
     }
 }
 
+const keepScanning = (input: string, start: number): boolean => {
```

Comment:
Docs would be helpful here for easier grokking. Rough suggestion for what would've helped me: 
```suggestion
/*
 * keepScanning tells ScanBalancedPattern whether to keep scanning for matching parentheses.
 *
 * Returns false if the input contains a valid filter or operator, in which case we don't want to parse the parentheses in 
 * the input as a literal pattern.
 *
 */ 
const keepScanning = (input: string, start: number): boolean => {
```